### ISSUE TITLE
Print element variables to csv (or other ascii) file?

### ISSUE THREAD

**Issue by krisbakernnpp** [idaholab/moose#7140]
I can't seem to find a way to print element variables, like element-averaged VonMises stress, to a csv file.  The point is that I don't want to use paraview to view contour plots, but rather print specific element variable values to compare against other codes more directly.  Printing quadrature point values would be nice as well, but I only saw one solution which is to create an aux variable for each quadrature point.  Is printing element variables within a specified block to csv files an option?

**Comment by permcody** [comment]
Well you could do this but why? You'd have to order them in some way or
describe their locations and then you're halfway to reinventing some new
proprietary mesh format. If you some ascii format you have options such as
XDA (libMesh's proprietary format). However, working with Exodus isn't that
bad. There's a small library that you can download and compile and you can
read information out of the file in a logical way.

Basically, if you describe what your needs are, we could probably get you
to a solution faster than telling you how to do something that might start
you on a longer road.

**Comment by friedmud** [comment]
In addition what @permcody mentioned there is also the fact that you're working with unstructured mesh... so there will be no ordering to the data... MOOSE can even re-order your elements before, during and after the simulation.

However, since this is a fairly typical activity there are a couple of things you can do:

1.  You already mentioned outputting the values of one qp into an Auxiliary field... people use that a lot.  And then (like @permcody mentioned) you can use Exodus utilities or Paraview to pull values out that you want.

2.  Use a `VectorPostprocessor`.  A `VectorPostprocessor` outputs values in a CSV file for each time step.  We already have one that can output the values of MaterialProperties at each quadrature point for each element along a line through the domain.
  Here is where the code is:
  https://github.com/idaholab/moose/blob/devel/framework/include/vectorpostprocessors/LineMaterialRealSampler.h
  And here is an example of using it:
  https://github.com/idaholab/moose/blob/devel/test/tests/vectorpostprocessors/line_material_sampler/line_material_real_sampler.i

  Note that there is a base class: 
  https://github.com/idaholab/moose/blob/devel/framework/include/vectorpostprocessors/LineMaterialSamplerBase.h
  
  So if you want to make your own kind of LineMaterialSampler (to sample your own kind of Material Properties) you can do that.

**Comment by permcody** [comment]
Also, asking for help should be done on the mailing list. Only a few people watch the issues:
https://groups.google.com/forum/#!forum/moose-users

**Comment by jessecarterMOOSE** [comment]
That's probably my fault. I told my colleagues to join Github so they could keep track of #7076 but didn't mention the google groups forum.

**Comment by friedmud** [comment]
No problem guys - we're just getting started on working together here... we'll work out the right channels of communication!

@krisbakernnpp please feel free to email the mailing list to continue this discussion... there are tons of people on there that need to compare solutions (especially solid-mechanical solutions) to other codes and I'm sure they will chime in with best practices...

**Comment by krisbakernnpp** [comment]
Thanks guy.  I'll do that.
